### PR TITLE
SDIT-1313 PersonResource

### PR DESCRIPTION
**Pull request by steverendell**
Enforce VIEW_CONTACTS role for /persons endpoints

### CODE REVIEW

File: src/main/java/uk/gov/justice/hmpps/prison/api/resource/PersonResource.java [hunk 1]
```diff
@@ -32,23 +32,13 @@ public PersonResource(final PersonService service) {
...
-        @ApiResponse(responseCode = "200", description = "OK"),
-        @ApiResponse(responseCode = "400", description = "Invalid request.", content = {@Content(mediaType = "application/json", schema = @Schema(implementation = ErrorResponse.class))}),
-            @ApiResponse(responseCode = "404", description = "Requested resource not found.", content = {@Content(mediaType = "application/json", schema = @Schema(implementation = ErrorResponse.class))}),
-            @ApiResponse(responseCode = "500", description = "Unrecoverable error occurred whilst processing request.", content = {@Content(mediaType = "application/json", schema = @Schema(implementation = ErrorResponse.class))})})
-    @Operation(summary = "The most recent value of each type of person identifier", description = "The most recent value of each type of person identifier")
-    @GetMapping("/{personId}/identifiers")
-    public List<PersonIdentifier> getPersonIdentifiers(@PathVariable("personId") @Parameter(description = "The persons NOMIS identifier (personId).", required = true) final Long personId) {
-        return service.getPersonIdentifiers(personId);
-    }
-
```
Comment (steverendell) [inline src/main/java/uk/gov/justice/hmpps/prison/api/resource/PersonResource.java:45]: unused

File: src/test/java/uk/gov/justice/hmpps/prison/api/resource/impl/ImageResourceIntTest.kt [hunk 2]
```diff
@@ -135,7 +135,6 @@ class ImageResourceIntTest : ResourceTest() {
     fun putImageReturnsForbiddenForIncorrectRoles() {
       webTestClient.post().uri("/api/images/offenders/A1234AA")
         .headers(setClientAuthorisation(listOf("ROLE_WRONG")))
-        .header("Content-Type", MediaType.MULTIPART_FORM_DATA_VALUE)
```
Comment (steverendell) [inline src/test/java/uk/gov/justice/hmpps/prison/api/resource/impl/ImageResourceIntTest.kt:138]: YADNI!

**Review by mikehalmamoj — APPROVED** [review]

File: src/main/java/uk/gov/justice/hmpps/prison/api/resource/PersonResource.java [hunk 3]
```diff
@@ -32,6 +33,18 @@ public PersonResource(final PersonService service) {
         this.service = service;
     }
 
+    @ApiResponses({
+        @ApiResponse(responseCode = "200", description = "OK"),
+        @ApiResponse(responseCode = "400", description = "Invalid request.", content = {@Content(mediaType = "application/json", schema = @Schema(implementation = ErrorResponse.class))}),
+            @ApiResponse(responseCode = "404", description = "Requested resource not found.", content = {@Content(mediaType = "application/json", schema = @Schema(implementation = ErrorResponse.class))}),
+            @ApiResponse(responseCode = "500", description = "Unrecoverable error occurred whilst processing request.", content = {@Content(mediaType = "application/json", schema = @Schema(implementation = ErrorResponse.class))})})
+    @Operation(summary = "The most recent value of each type of person identifier", description = "The most recent value of each type of person identifier")
+    @PreAuthorize("hasRole('ROLE_VIEW_CONTACTS')")
```
Comment (karenmillermoj) [inline src/main/java/uk/gov/justice/hmpps/prison/api/resource/PersonResource.java:42]: All looks good - but not sure about the role

**Review by karenmillermoj — APPROVED** [review]
lgtm

**Pull request merged**